data_structures/binary_tree/morris_preorder_traversal.py

```diff
@@ -0,0 +1,168 @@
+"""
+Problem Statement: Given a binary perform an preorder traversal using Morris Preorder
+traversal algorithm. (Iterative version of Preorder traversal of tree)
+
+https://www.geeksforgeeks.org/morris-traversal-for-preorder/
+"""
+
+
+class TreeNode:
+    """
+    Class representing a node in a binary tree.
+
+    Attributes:
+    -----------
+    value : int
+        The value stored at the node.
+    left : TreeNode
+        Pointer to the left child node (default is None).
+    right : TreeNode
+        Pointer to the right child node (default is None).
+    """
+
+    def __init__(self, value: int) -> None:
+        self.value = value
+        self.left: TreeNode | None = None
+        self.right: TreeNode | None = None
+
+
+class BinaryTree:
+    """
+    Class representing a binary tree.
+
+    Methods:
+    --------
+    insert(value: int) -> None:
+        Insert a value into the binary tree following binary search tree (BST) rules.
+
+    morris_preorder_traversal() -> List[int]:
+        Perform preorder traversal and return list of node values.
+
+
```

Comment:
```suggestion
```
I don't think it's necessary to specify the methods in the docstring because the methods themselves should already be documented anyway.